### DISCUSSION TITLE
"placeholderData: keepPreviousData" results in double fetch

### DISCUSSION THREAD

**Discussion by skelawsky** [TanStack/query#7128] [Q&A]
@HugeLetters 
I experienced the same error as described in the topic and tried to apply your solution however another problem arose: the query is fetched twice on initial render even though the query key does not change and it only happens when I add `placeholderData: prevData`. Have you had the same problem??

```
	let prevData;
	$: documentLinesQuery = createQuery({
		queryKey: ['document-lines', perPage, page, $sorting, filters],
		queryFn: async () => {
			console.log('fetch')
			let url = new URL(`document-lines`, PUBLIC_API_URL);
			appendSearchParams({ url, perPage, page, sorting: $sorting, filters });
			return await fetch(url).then((res) => res.json());
		},
		placeholderData: prevData
	});
	$: if ($documentLinesQuery?.isSuccess) {
		prevData = $documentLinesQuery.data;
	}
```
![image](https://github.com/TanStack/query/assets/75502784/32e6c2f3-75cb-4b1a-b5a3-0a20a341c235)

_Originally posted by @skelawsky in https://github.com/TanStack/query/issues/5913#issuecomment-1978149433_

I'm creating a new post because linked one seems to be closed and not visible to the others.

**Comment by HugeLetters** [comment]
I don't remember if I had the same issue - I only found out placeholdeData not working properly because I was playing around with svelte-query, not because I had a problem to solve with it

As to why it happens - I would guess it's because query is recreated when you set new placeholderData. Perhaps you could try to set some other option the same way and see if you have the same issue - if yes I bet the problem is because query is recreated.
